src/GrafanaDashboardDeploy/src/GrafanaClient.cs

```diff
@@ -0,0 +1,327 @@
+                    {
+                        folderResponse = await JObject.LoadAsync(jr).ConfigureAwait(false);
+                    }
+                }
+            }
+
+            return folderResponse;
+        }
+
+        public async Task<JObject> CreateDatasourceAsync(JObject datasource)
+        {
+            var uri = new Uri(new Uri(BaseUrl), "/api/datasources");
+            JObject responseJson;
+
+            var stream = new MemoryStream();
+            using (var textWriter = new StreamWriter(stream))
+            using (var jsonStream = new JsonTextWriter(textWriter))
+            {
+                var serializer = new JsonSerializer();
+                serializer.Serialize(jsonStream, datasource);
+
+                jsonStream.Flush();
+                stream.Position = 0;
+
+                using (var content = new StreamContent(stream))
+                {
+                    content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+                    HttpResponseMessage response = await _client.PostAsync(uri, content).ConfigureAwait(false);
+
+                    if (!response.IsSuccessStatusCode)
+                    {
+                        _logger.LogWarning("Response does not indicate success");
+                        _logger.LogWarning("Status {}: {}", response.StatusCode, response.ReasonPhrase);
+                        _logger.LogDebug(await response.Content.ReadAsStringAsync().ConfigureAwait(false));
+
+                        // TODO: How to handle error?
+                    }
+
+                    var st = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                    using (var sr = new StreamReader(st))
+                    using (var jr = new JsonTextReader(sr))
+                    {
+                        responseJson = await JObject.LoadAsync(jr).ConfigureAwait(false);
+                    }
+                }
+            }
+
+            return responseJson;
+        }
+
+        public async Task<JObject> CreateDashboardAsync(JObject dashboard, int folderId)
+        {
+            var uri = new Uri(new Uri(BaseUrl), "/api/dashboards/db");
+
+            var dashboardBody = new JObject(
+                new JProperty("dashboard", dashboard),
+                new JProperty("folderId", folderId),
+                new JProperty("overwrite", false));
+
+            JObject responseJson;
+
+            var stream = new MemoryStream();
+            using (var textWriter = new StreamWriter(stream))
+            using (var jsonStream = new JsonTextWriter(textWriter))
+            {
+                var serializer = new JsonSerializer();
+                serializer.Serialize(jsonStream, dashboardBody);
+
+                jsonStream.Flush();
+                stream.Position = 0;
+
+                using (var content = new StreamContent(stream))
+                {
+                    content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
```

Comment:
Might jsut want to plop this in the HttpClient.DefaultHeaders, since every request/response is json (the "Accept" should probably be json too)